### PR TITLE
Add full tests to publish action.

### PR DESCRIPTION
**Pull request by h-mayorquin**
As described here, doing from main branch to test this.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
OK, the tets are faling when the version iends in `.dev` as it should.

https://github.com/SpikeInterface/spikeinterface/actions/runs/5599159488/jobs/10239760746?pr=1854

Let's see if the tests are working as they should.

File: .github/workflows/publish-to-pypi.yml [hunk 1]
```diff
@@ -4,32 +4,46 @@ on:
...
+      #     is_dev_version=$(python -c "import importlib.metadata; package_name = 'spikeinterface'; version = importlib.metadata.version(package_name); print(version.endswith('dev0'))")
+      #     if [ $is_dev_version = "True" ]; then
+      #       echo "Dev version detected,remember to fix the version before the release!"
+      #       exit 1
+      #     fi
+      - name: Test everything but external sorters before release
         run: |
-          python -m build --sdist --wheel
-          twine upload dist/*
+          pytest -m "not sorters_external"
```
Comment (h-mayorquin) [inline .github/workflows/publish-to-pypi.yml:42]: I am keping this as in the full tests with code coverage but I don't know if we should test EVERYTHING here.

**Comment by h-mayorquin** [comment]
OK, so the full tests are running reasonably:
https://github.com/SpikeInterface/spikeinterface/actions/runs/5599505029/jobs/10240551443?pr=1854

I think this is ready to go. The only thing left is that if it feals when you push the new tag you should fix the tests and then do a manual run of this action.

File: .github/workflows/publish-to-pypi.yml (hunk 1 above)
Comment (alejoe91) [reply .github/workflows/publish-to-pypi.yml:40]: fair to me

**Comment by alejoe91** [comment]
I'm ok with this! Let's wait to hear @samuelgarcia thoughts.

Another thing we should do is to run all tests if the `pyproject.toml` is changed (basically consider it as a core change)

**Comment by samuelgarcia** [comment]
I am not sure to be happy with this.
the idea was:
* we prepare a branch to prepare the release. In this branch there not "dev" and no neo+probeinterface githib install 
   because we comment
* this branch should trigger normally all test, this is not the case unfortunatly.
* Then push to pypi is quite fast and make only on core test on one sytsem (just in case).

But for me the entire test are triggered and check in the prepare relase PR not before pushing to pypi.
Otherwise we have to wait too long.
Lets have a call soon baout this.

**Comment by h-mayorquin** [comment]
This is better done with a combination of :
https://github.com/SpikeInterface/spikeinterface/pull/1856
And:
https://github.com/SpikeInterface/spikeinterface/pull/1858

Closing this.

**Pull request closed without merging**